### DISCUSSION TITLE
Why do exceptional files exist?

### DISCUSSION THREAD

**Discussion by xnuk** [pi0/config-dir#11] [General]
If `.env` and `.git*` files don't fit in the `.config/` folder, then what's the specific reason of that?

### Some thoughts
- [Some guide (12factor.net) suggests to use env vars as a config](https://12factor.net/config).
- Some uses `.env.development` or `.env.example` in the project's root. Are that exceptional files too?
- Git-related files can be put in `.config/git/`. [Git already reads `~/.config/git/ignore` and `.git/info/exclude`, not just `.gitignore`](https://git-scm.com/docs/gitignore). It's matter of choice I think.
- Or, in radical way, could just move `.git` into `.config/git/` (and change `GIT_DIR` environment to workaround for now).

**Comment by pi0** [comment]
.git is a storage directory. And i tend to agree with you about 12 factor recommendations over physical file for .env. But it would make it a long discussion here to even involve and can distract proposal from its main goals that’s why for now explicitly excluded.
